pkg/common-controller/snapshot_controller_base.go

```diff
@@ -42,6 +42,12 @@ import (
 	"k8s.io/kubernetes/pkg/util/goroutinemap"
 )
 
+// Number of retries when we create a snapshot content object for a snapshot.
+const createSnapshotContentRetryCount = 5
```

Comment:
my higher level question is why do we need to retry within the workqueue func? Can we just re-add the snapshot to the queue for retry whenever there's a failure, and let the workqueue properly throttle the retries?

Reply:
I thought this is recommended practice for API update?  PV controller also retries when PV object fails to be created.  I remember when I first started to work on snapshot in external-storage repo, I was asked to add this type of retry as a bug fix.

Reply:
I believe PV controller does it because of the possibility of leaking the volume. I don't think we have the same issue here since we are creating the content object first before calling CSI. Cc @jsafrane

Reply:
> I don't think we have the same issue here since we are creating the content object first before calling CSI

+1, in PV controller, a volume is already created on storage backend when the controller saves a PV. And it really needs to save it, otherwise the volume may be leaked.

In snapshots, nothing is leaked if SnapshotContent is not created, right?

Reply:
Yes, in the beta version, we changed the behavior and create the VolumeSnapshotContent first before creating the physical snapshot resource.

We are also doing retries when update snapshot fails.  In general, is it recommended practice to do retries when update an API object fails?

Reply:
The general pattern is to requeue it on any failure and let workqueue schedule the retry.

Reply:
Removed the RetryCount.